server/resp/src/main/java/org/infinispan/server/resp/filter/EventListenerKeysFilter.java

```diff
@@ -42,12 +36,20 @@ public EventListenerKeysFilter(byte[][] keys) {
       }
    }
 
+   public EventListenerKeysFilter(byte[] key) {
+      List<byte[]> list = new ArrayList<>(1);
+      list.add(key);
+      this.keys = Map.of(key.length, list);
+   }
+
+   @ProtoFactory
    EventListenerKeysFilter(List<byte[]> keys) {
       this(keys.toArray(Util.EMPTY_BYTE_ARRAY_ARRAY));
```

Comment:
I would use this constructor everywhere and get rid of `public EventListenerKeysFilter(byte[][] keys)`. It seems the RESP server code invokes `List.toArray()` every time this filter is used.

Reply:
I have updated it so that we always have `public EventListenerKeysFilter(Stream<byte[]> keys) {` except in the case of a single key.